src/pages/Events/Events.tsx

```diff
@@ -16,14 +17,20 @@ const Events = () => {
   const [buttonVariants, setButtonVariants] = useState<boolean[]>([]);
   const [showPopup, setShowPopup] = useState<boolean>(false);
   const [index, setIndex] = useState<number>(-1);
+  const [date, setDate] = useState<string>("");
```

Comment:
I do not think that you need to store this in state. I will add more notes on this below by your useEffect.